pkg/http/listener_test.go

```diff
@@ -0,0 +1,819 @@
+/*
+ * Minio Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package http
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/minio/minio-go/pkg/set"
+)
+
+var serverPort uint32 = 60000
+
+func getNextPort() string {
+	port := atomic.LoadUint32(&serverPort)
+	atomic.AddUint32(&serverPort, 1)
+	return strconv.Itoa(int(port))
```

Comment:
AddUint32() return the new value, so line 41 can be collapsed into line 42 as follows:
```
port := atomic.AddUint32(&serverPort, 1) - 1
```